### DISCUSSION TITLE
Modify Frustum dynamically

### DISCUSSION THREAD

**Discussion by peiyanlu** [iTwin/itwinjs-core#6758] [Q&A]
Now, I have a use case, I load two .bim files in one viewport, two iModels means two projectExtents, if I load a small range of iModels first, the other iModel will be cropped by the frustum when browsing, if I load a large range of iModels first, the other iModel will be loaded and browse normally, so I want to resize the frustum to ensure normal browsing, but it doesn't work, I do this:
```ts
  const range1 = vp.iModel.projectExtents
  const range2 = iModel.projectExtents
  range1.extendRange(range2)
  
  const frustum = vp.getFrustum()
  frustum.setFrom(Frustum.fromRange(range1))
  vp.setupViewFromFrustum(frustum)
  vp.changeView(vp.view)
  
  vp.view.lookAtVolume(range1, vp.viewRect.aspect)
  vp.synchWithView({ animateFrustumChange: true })
```

**Comment by pmconne** [comment]
The frustum is recomputed every time you perform any view navigation.

I think you need to reconsider why you are attempting to view multiple iModels.

**Reply by peiyanlu** [reply]
We hope to be able to freely combine browsing without having to merge the original files every time. Currently, iTwin provides a way to load additional iModels, and we hope to use it to improve browsing efficiency, but in some aspects, it is not perfect enough

**Reply by pmconne** [reply]
> iTwiin provides a way to load additional iModels

No, it doesn't. There are hacks you can use to get more than one iModel displaying, which only come close to working well if you're viewing two versions of the same iModel. In any other scenario, it will not work as you expect.

If you're making one iModel per DGN file or IFC file or whatever, please rethink your approach.

**Reply by peiyanlu** [reply]
Thank you for your reply. I will rethink it. Thank you again
